BentoKit/BentoKit/Screen/ScreenLifecycleEvent.swift

```diff
@@ -0,0 +1,35 @@
+public protocol ScreenLifecycleObserving {
+    func send(_ event: ScreenLifecycleEvent)
+}
+
+public struct ScreenLifecycleEvent {
+    public let status: Status
+    public let isBeingPresentedInitially: Bool
+    public let isBeingRemovedIndefinitely: Bool
+
+    public init(status: Status,
+                isBeingPresentedInitially: Bool,
+                isBeingRemovedIndefinitely: Bool) {
+        self.status = status
+        self.isBeingPresentedInitially = isBeingPresentedInitially
+        self.isBeingRemovedIndefinitely = isBeingRemovedIndefinitely
+    }
+
+    public enum Status {
```

Comment:
I wonder if it's enough just to have it as an `enum` with associated values? Instead of `Status + Event`

Reply:
👍 especially as the `isBeing...` flags don't apply to each situation.

Reply:
Having `viewDidLoad` would be nice as well

Reply:
```swift
enum Event {
    case viewDidLoad
    case viewWillAppear(isInitial: Bool)
    case viewDidAppear(isInitial: Bool)
    case viewWillDisappear(isFinal: Bool)
    case viewWillDisappear(isFinal: Bool)
}
```

seems sufficient as already mentioned.